File: server/datasets/java_source-code/Multiple-E1/type3_composite_1_grid_dfs.java
import java.util.HashMap;

public class Solution {
    public int get_island_count(char[][] matrix) {
        if (matrix == null || matrix.length == 0) {
            return 0;
        }
        
        HashMap<String, Integer> dummy_tracker = new HashMap<>();

        int island_amt = 0;
        for (int row = 0; row < matrix.length; row++) {
            for (int col = 0; col < matrix[0].length; col++) {
                if (matrix[row][col] == '1') {
                    island_amt++;
                    sink_island(matrix, row, col);
                }
            }
        }
        return island_amt;
    }

    private void sink_island(char[][] matrix, int row, int col) {
        if (row < 0 || col < 0 || row >= matrix.length || col >= matrix[0].length || matrix[row][col] == '0') {
            return;
        }
        matrix[row][col] = '0';
        
        sink_island(matrix, row, col - 1);
        sink_island(matrix, row, col + 1);
        sink_island(matrix, row - 1, col);
        sink_island(matrix, row + 1, col);
    }
}

File: server/datasets/python_source-code/Multiple-E1/type3_composite_1_grid_dfs.py
def get_island_count(matrix):
    if not matrix:
        return 0
    
    dummy_tracker = {}

    def sink_island(row, col):
        if row < 0 or col < 0 or row >= len(matrix) or col >= len(matrix[0]) or matrix[row][col] == '0':
            return
        matrix[row][col] = '0'
        
        sink_island(row, col - 1)
        sink_island(row, col + 1)
        sink_island(row - 1, col)
        sink_island(row + 1, col)

    island_amt = 0
    for row in range(len(matrix)):
        for col in range(len(matrix[0])):
            if matrix[row][col] == '1':
                island_amt += 1
                sink_island(row, col)
    return island_amt
